core/src/test/java/com/linecorp/armeria/common/util/CompositeExceptionTest.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.util;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.internal.testing.AnticipatedException;
+
+class CompositeExceptionTest {
+
+    private static final String separator = System.getProperty("line.separator");
+    private static final String PADDING = "  ";
+
+    @Test
+    void verboseExceptionEnabledTest() {
+        final IllegalStateException ex1 = new IllegalStateException();
+        final IllegalArgumentException ex2 = new IllegalArgumentException();
+        final CompositeException compositeException =
+                new CompositeException(ImmutableList.of(ex1, ex2), Sampler.always());
+        final int separatedStacktraceLength =
+                (int) Arrays.stream(compositeException.getCause().getMessage().split(separator))
+                        .filter(line -> !line.contains("Multiple exceptions") && !line.contains("|-- "))
+                        .count();
+
+        final int sumStackTraceOutputLength = ex1.getStackTrace().length + ex2.getStackTrace().length;
+        // if verboseException option enabled, output full stacktrace
+        // this case is occurred 2 exceptions (ex1 stacktrace length + ex2 stacktrace length)
+        assertThat(separatedStacktraceLength).isEqualTo(sumStackTraceOutputLength);
+    }
+
+    @Test
+    void verboseExceptionDisabledTest() {
+        final IllegalStateException ex1 = new IllegalStateException();
+        final IllegalArgumentException ex2 = new IllegalArgumentException();
+        final CompositeException compositeException =
+                new CompositeException(ImmutableList.of(ex1, ex2), Sampler.never());
+        final int separatedStacktraceLength =
+                (int) Arrays.stream(compositeException.getCause().getMessage().split(separator))
+                        .filter(line -> !line.contains("Multiple exceptions") && !line.contains("|-- "))
+                        .count();
```

Comment:
I have seen duplicate code of this logic. How about extracting it so that we can use it in multiple places?
```java
private static long stackTraceLength(CompositeException compositeException) {
    return Arrays.stream(compositeException.getCause().getMessage().split(separator))
                 .filter(line -> !line.contains("Multiple exceptions") && !line.contains("|-- "))
                 .count();
}
```
Please, also, note that we don't need to downcast to integer and the `T` in `stackTrace` is capitalized.